byway/src/main/webapp/routepage.js

```diff
@@ -12,40 +12,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-
-// copies of recommendations and selected stops, stored as arrays for synchronous updating
-let recs = [];
-let stops = []; // TODO: Make set to avoid duplicates
-
+ 
+// copies of recommendations and selected stops, stored as sets for synchronous updating
+let recs = new Set();
+let stops = new Set();
+ 
+// location representations of stops array
+let waypoints = new Set();
+ 
+// object that communicates with the GMaps API service
+let directionsService;
+ 
+// object that renders display results on the map
+let directionsRenderer;
+ 
+// TODO: get from Trip key
+let start;
+let end;
```

Comment:
Are start and end also part of `stops` or no?

Reply:
I will take it out of the stops bc it will map the place as both a waypoint and as start point. Also start == end for the roadtrip so I think I will include other major cities as waypoints.